### PR TITLE
Add optional usage_active and time_active CPU metrics

### PR DESCRIPTION
**Pull request by bobmshannon**
An approach for solving #2710. Uses the existing code, but adds two new metrics with the assumption that `%busy=100-%idle`, and `time busy=total time-time idle`.

### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

### CODE REVIEW

**Comment by phemmer** [comment]
My $0.02:
* I personally dislike computed fields being emitted. Computed fields, especially ones that are cheap to compute, just take up space in the database, and reduce its performance.
* I think the "busy" field would be confusing. Because we're simply doing `busy=100-idle`, things like steal time, and iowait, end up as "busy". This really muddies the definition of what "busy" is. With steal time, busy would be high, but the CPU time isn't being spent inside the OS being monitored. With iowait, busy would be high, but the CPU is really idle as it can be scheduled to run other tasks while the IO operations are outstanding, so it's not really busy.

**Comment by bobmshannon** [comment]
@phemmer Good points. What would your thoughts be on tweaking the formula for busy to be `busy=100-idle-iowait`, and documenting that this field is measuring meaningful work performed by the CPU?

The first point about computed fields is also a good one, and I'm not sure if this is something that is typically avoided in other input plugins as a matter of style/convention/agreed upon best practice.

**Comment by danielnelson** [comment]
In general I agree with @phemmer, we should not have values that are computed based on other fields.  However, this metric has been frequently requested since it's removal, since it is the most user friendly way of viewing cpu use.  This is why I think we should probably make an exception here.

Maybe we should take a page from collectd and add an "active" state if a `report_active` option is set to true?  The active state would be defined in the same way collectd defines it.  The default value would be false.  https://collectd.org/documentation/manpages/collectd.conf.5.shtml#plugin_cpu

**Comment by bobmshannon** [comment]
I like the configurable option (false by default?) because it allows someone optionally to trade off a small amount of overhead for the convenience of having this computed field.

Anyway, I've updated this PR accordingly to reflect that approach and have also addressed the iowait issue mentioned in the comments above.

File: plugins/inputs/system/cpu.go [hunk 1]
```diff
@@ -116,6 +129,11 @@ func totalCpuTime(t cpu.TimesStat) float64 {
 	return total
 }
 
+func activeCpuTime(t cpu.TimesStat) float64 {
+	active := totalCpuTime(t) - t.Idle - t.Iowait
```
Comment (danielnelson) [inline plugins/inputs/system/cpu.go]: Don't subtract Iowait, to match collectd.

**Comment by bobmshannon** [comment]
@danielnelson the iowait subtraction issue should be addressed

**Comment by danielnelson** [comment]
Thanks!

**Comment by rwarren** [comment]
Is the new `time_active` any different than `(100 - usage_idle)`?

**Pull request merged**